src/Data/Collector.ts

```diff
@@ -270,6 +275,87 @@ const isReceiptRobust = async (
   return { success: true }
 }
 
+// Offline receipt verification
+const verifyReceiptOffline = async (
+  receipt: Receipt.ArchiverReceipt, 
+  executionGroupNodes: ConsensusNodeInfo[],
+  minConfirmations: number
+): Promise<{ success: boolean; newReceipt?: Receipt.ArchiverReceipt }> => {
+  const { signedReceipt, globalModification } = receipt
+
+  if (globalModification) {
```

Comment:
it will be more readable if we have refactor this method to have two more function invoked based on this flag; `verifyGlobalTxReceiptOffline` and `verifyNonGlobalTxReceiptOffline`

Reply:
we could probably call it internally instead of calling it from the main function

Reply:
we could probably call it internally for different methods instead of calling it twice from the main function

Reply:
done